tasks/insights-client.yml

```diff
@@ -36,6 +36,32 @@
     insertafter: "#auto_update"
     line: auto_update={{ rhc_insights.autoupdate | d(true) | bool }}
 
+- name: Configure ansible host
+  block:
+    - name: Configure ansible host in config
+      lineinfile:
+        path: "{{ __rhc_insights_conf }}"
+        regexp: "^ansible_host"
+        insertafter: "#ansible_host"
+        line: ansible_host={{ rhc_insights.ansible_host }}
+      when: 
+        - rhc_insights.ansible_host is defined
+        - rhc_insights.ansible_host != None
+
+    - name: Attempt ansible host name update in inventory
+      vars:
+        host: "{{ rhc_insights.ansible_host }}"
+      shell: insights-client --ansible-host {{ host }} & wait
+      when:
+        - >-
+          "This host is registered" in __rhc_insights_status.stdout
+          or "Registered" in __rhc_insights_status.stdout
+        - >-
+          rhc_insights.ansible_host is defined
+          and rhc_insights.ansible_host != None
+      register: __rhc_insights_ansible_host_change_result
+      changed_when: true
```

Comment:
Can this be made idempotent?  That is, if `rhc_insights.ansible_host` is already the value of `insights-client --ansible-host`, it would be better to return `changed_when: false` in that case

Reply:
I'm actually not sure about this because I don't think there's a way to check what the current `ansible_host` is from inventory. The `insights-client --ansible-host` command expects an input and updates inventory even if the same host name is provided as the one in inventory. Leaving the input param blank does not return the current host name in inventory afaik.

Reply:
ok - @ptoscano ^^^ ?

Reply:
> I'm actually not sure about this because I don't think there's a way to check what the current `ansible_host` is from inventory.

Theoretically it could be done by querying the API manually, however that would add a gigantic amount of complications -> better not.

So I think the better way is to always set the hostname...

Reply:
I think this could be simplify. It is no need to use the `--ansible-host` parameter in the second task. Once you set the variable it will update the ansible-host name in inventory in the next collection.

Other way to achieve this is to use the parameter, and it will update in inventory. But it not keep tracks in the config file. That means the next collection will not use the new ansible-host name and will update to the one that is in configuration.

So I will update the config file, and make it idempotent checking with the ansible-host defined there.

Reply:
How important is it for inventory to be immediately updated? If the ansible host name in inventory is not required to be immediately updated when using the role, then Alba's idea of updating only the config and waiting for the next collection would be a much simpler change.

Any further thoughts on this?

Reply:
I think the general idea is that, in case you change something, that something is immediately reflected. I think it would be awkward otherwise that you change `ansible_host`, run the playbook/role, and there is no change for the system until the next check-in. Also, applying the change immediately is what is done already for tags -- see the `Do the collection if insights-client was registered` task.

So this would be yet another condition to trigger the manual collection.

Reply:
Agreed ^^^

I've pushed an update to simplify the yaml and added a condition to check for changes in the ansible host to trigger a manual  collection [[79e1bb8](https://github.com/linux-system-roles/rhc/pull/155/commits/79e1bb818dedfb1a0d91c9c26c76c9f94b39fa99)]. 
I hope that addresses everything in this thread.